### ISSUE TITLE
Check and fix tests

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/guarana#1]
Current tests in the stable-4.8 branch produces the output as below. Should be switched to `TestDirectory`, checked and polished.

```
============================OUTPUT START==============================
#I  RunPackageTests("guarana", "0.94", "tst/testall.g", auto);
 *********   GAP unknown of 2017-09-21 22:16:46 (UTC)
 *  GAP  *   https://www.gap-system.org
 *********   Architecture: x86_64-pc-linux-gnu-gcc-default64
 Libs used:  gmp, readline
 Loaded workspace: wsp.g
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   AClib 1.2, Alnuth 3.0.0, AtlasRep 1.5.1, AutPGrp 1.8, 
             Browse 1.8.7, CRISP 1.4.4, Cryst 4.1.12, CrystCat 1.1.6, 
             CTblLib 1.2.2, FactInt 1.5.4, FGA 1.3.1, GAPDoc 1.6, 
             Guarana 0.94, IO 4.4.6, IRREDSOL 1.4, LAGUNA 3.7.0, nq 2.5.3, 
             Polenta 1.3.7, Polycyclic 2.11, RadiRoot 2.7, ResClasses 4.6.0, 
             Sophus 1.23, SpinSym 1.5, TomLib 1.2.6, utils 0.48
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
#I  ReadTest is no longer supported. Please use more robust and flexible
#I  Test. For backwards compatibility, ReadTest(<filename>) is replaced
#I  by Test( <filename>, rec( compareFunction := "uptowhitespace" ))
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 22:
# Input is:
h := RandomGrpElm( mo );
# Expected output:
[ -8, 1, 3, -7, 0 ]
# But found:
[ -1, -3, -7, 0, 9 ]
########
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 24:
# Input is:
y := RandomLieElm( mo );
# Expected output:
[ 7, 6, -9, -9, -2 ]
# But found:
[ -6, 8, 7, -8, -6 ]
########
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 30:
# Input is:
k := Exp( y );
# Expected output:
[ 7, 6, 12, -2, 89/2 ]
# But found:
[ -6, 8, -17, 31, -94 ]
########
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 34:
# Input is:
g*h;
# Expected output:
[ -7, 2, -5, 31, -17/2 ]
# But found:
[ 0, -2, -8, 3, 23/2 ]
########
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 36:
# Input is:
Comm(g,h);
# Expected output:
[ 0, 0, -9, 33, -12 ]
# But found:
[ 0, 0, 2, 8, 7 ]
########
########> Diff in /home/gap/inst/gap-stable-4.8/pkg/guarana/tst/guarana1.tst, \
line 38:
# Input is:
Comm(x,y);
# Expected output:
[ 0, 0, 11, -8, 13/2 ]
# But found:
[ 0, 0, -16, 21/2, -14 ]
########
Test 1 of guarana package
GAP4stones: 312
#I  ReadTest is no longer supported. Please use more robust and flexible
#I  Test. For backwards compatibility, ReadTest(<filename>) is replaced
#I  by Test( <filename>, rec( compareFunction := "uptowhitespace" ))
Test 2 of guarana package
GAP4stones: 277
#I  RunPackageTests("guarana", "0.94", "tst/testall.g", auto): runtime 392
============================OUTPUT END================================
```

**Comment by olexandr-konovalov** [comment]
Update: now one can see diffs in the package tests when they are run with the `stable-4.9` branch of the GAP repository. Please see https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9-staging to inspect them: click on the line corresponding to this package and scroll till the very end of the log file to see the output of the package tests, the summary, and further instructions.

As soon as the package tests will run without diffs, they will be moved to https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9. This will be very helpful to monitor in the future that the package is still usable and that no changes in the GAP system break the package.

**Comment by fingolfin** [comment]
This was resolved by PR #2
